zerver/views/__init__.py

```diff
@@ -409,9 +409,7 @@ def maybe_send_to_registration(request, email, full_name=''):
             # Explanation: http://stackoverflow.com/a/5605354/90777
             urllib.parse.quote_plus(full_name.encode('utf8')))))
     else:
-        return render_to_response('zerver/accounts_home.html',
-                                  {'form': form, 'current_url': request.get_full_path},
-                                  request=request)
+        return redirect('/accounts/home/?email=' + email)
```

Comment:
The old code was specifying wrong action in the form. In my case the action was the complete url for GitHub oAuth. As a result, submitting the form returned error.